### PR TITLE
ferium 4.1.10 (new formula)

### PR DESCRIPTION
**Pull request by theRookieCoder**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Adds [ferium](https://github.com/gorilla-devs/ferium)

### CODE REVIEW

**Comment by theRookieCoder** [comment]
Does anyone know how to install completions that are created from the command itself? Ferium outputs completions to stdio using the `complete` subcommand, how would this be installed using `<shell>_completion.install`?

File: Formula/ferium.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  on_linux do
+    depends_on "pkg-config" => :build
+  end
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system "ferium", "help"
```
Comment (chenrui333) [inline Formula/ferium.rb:20]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests.

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

File: Formula/ferium.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  depends_on "rust" => :build
+
+  on_linux do
+    depends_on "pkg-config" => :build
+  end
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
```
Comment (chenrui333) [inline Formula/ferium.rb:17]:
```suggestion

    bash_output = Utils.safe_popen_read(bin/"ferium", "complete", "bash")
    (bash_completion/"ferium").write bash_output
    zsh_output = Utils.safe_popen_read(bin/"ferium", "complete", "zsh")
    (zsh_completion/"_ferium").write zsh_output
    fish_output = Utils.safe_popen_read(bin/"ferium", "complete", "fish")
    (fish_completion/"ferium.fish").write fish_output
  end
```

File: Formula/ferium.rb (hunk 1 above)
Comment (theRookieCoder) [reply Formula/ferium.rb:27]: Hmmm that might be difficult because of gorilla-devs/ferium#87

Comment (theRookieCoder) [reply Formula/ferium.rb]: Done 5ee8ca928afbedecd5c9266d9677d97a6f51238d

btw these tests will fail if there is not internet connection, is that fine?

**Comment by theRookieCoder** [comment]
Oops I think I'm not supposed to merge. Should I revert that, or is this PR going to be squash merged anyways?

**Comment by Blakeinstein** [comment]
Why was this closed?

**Comment by theRookieCoder** [comment]
The PR got really messy with tons of commits, and I also accidently merged master.

**Pull request closed without merging**